### PR TITLE
snowflake freshness helper

### PR DESCRIPTION
**Pull request by dpeng817**
This PR introduces a standalone method to the dagster-snowflake package that, given a list of tables and a project ID, returns the freshness of each table.

How I tested this
Added a test that creates an observable source asset that attaches freshness as metadata to its observe result, and asserts the information is correct on the resulting observation event.

### CODE REVIEW

File: python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py [hunk 1]
```diff
@@ -713,3 +713,45 @@ def my_snowflake_job():
     return SnowflakeConnection(
         config=context, log=context.log, snowflake_connection_resource=snowflake_resource
     )
+
+
+def fetch_last_updated_timestamp(
```
Comment (sryza) [inline python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py:718]: This is operating on plurals, so I would expect it to be `timestamps`

File: python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py [hunk 2]
```diff
@@ -713,3 +713,45 @@ def my_snowflake_job():
...
+    *,
+    snowflake_connection: Union[SqlDbConnection, snowflake.connector.SnowflakeConnection],
+    schema: str,
+    tables: Sequence[str],
+) -> dict[str, float]:
```
Comment (sryza) [inline python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py:723]: Preference for Mapping over dict

**Review by sryza — COMMENTED** [review]
Thoughts on making this a method on `SnowflakeConnection`? It seems like that's what we've done with the other snowflake utilities.z

**Comment by dpeng817** [comment]
@sryza it's a bit confusing, but the `SnowflakeConnection` object we return from calling `SnowflakeResource.get_connection` is not the same as the one we define in code. From the documentation of our own SnowflakeConnection class:
```
Note that the SnowflakeConnection is only used by the snowflake_resource. The Pythonic SnowflakeResource does not use this SnowflakeConnection class.
```
So putting it on the `SnowflakeConnection` object isn't what we want, I think.

I considered putting it on the SnowflakeResource pythonic resource object as a class method, but that also seems weird, because we don't know the patterns of how users are going to open and close connections. It feels to me most composable to allow users to potentially open one connection, call out to this method to retrieve freshness, and potentially do some other stuff, all without needing to open multiple context managers, whereas a method on the resource would have to open it's own connection directly, irrespective of what connections might already be opened by the user.
Tl;dr, standalone function seems the most composable to me.

**Review by sryza — COMMENTED** [review]
Ah interesting. Yeah based on that I agree that a standalone method probably makes the most sense.

Curious for @jamiedemaria 's take. Jamie- is `dagster_snowflake.SnowflakeConnection` kind of essentially a deprecated API now that we endorse `SnowflakeResource`?

**Comment by jamiedemaria** [comment]
>  is dagster_snowflake.SnowflakeConnection kind of essentially a deprecated API now that we endorse SnowflakeResource?

yeah. We decided that with the pythonic snowflake resource we wanted to just provide snowflake's `SnowflakeConnection` (we'd gotten a lot of feature requests about the behavior in our `SnowflakeConnection`, and felt that giving direct access to the underlying snowflake.SnowflakeConnection would basically solve all of the issues)

I'm not sure we can formally mark it as deprecated though. I'm not sure if all old-style resources can be marked deprecated yet and we probably can't mark our `SnowflakeConnection` class deprecated until then

**Review by sryza — COMMENTED** [review]
The last thing I realized here is that this returns a float, which means that, if people just pass it through to the metadata dictionary, it will appear in the UI as a float. I think it would be much better for it to display as a timestamp, which probably means that this should return some sort of timestamp type.

**Comment by dpeng817** [comment]
@sryza I found this surprising, but we don't actually have a TimestampMetadataValue, as of now. I can add one though if we think it's important. To unblock the demo I wonder if that should be a fast follow.

**Comment by sryza** [comment]
Ah interesting. Yes I think worth adding one. For the demo timeframe, thoughts on having this API return datetimes and the demo code convert it to floats before passing it to the metadata dict? Then we wouldn't need to change this API later.

**Comment by dpeng817** [comment]
Good point. Momentarily forgot that this was only a utility function. I'll change to using datetimes.

**Review by sryza — APPROVED** [review]
I think the "Returns" section of the  docstring is out of date. Otherwise LGTM!
